daffodil-runtime1-unparser/src/main/scala/org/apache/daffodil/unparsers/runtime1/BinaryNumberUnparsers.scala

```diff
@@ -56,12 +58,8 @@ abstract class BinaryNumberBaseUnparser(override val context: ElementRuntimeData
     val nBits = getBitLength(state)
     val value = getNumberToPut(state)
     val dos = state.dataOutputStream
-    val res =
-      if (nBits > 0) {
-        putNumber(dos, value, nBits, state)
-      } else {
-        true
-      }
+    val res = putNumber(dos, value, nBits, state)
```

Comment:
Seems this should be
```scala
if (nBits == 0) UE(...)
else putNumber(...)
```
I think the original purpose of this bug was to make it so a length of 0 changes from being ignored to being an error. Does it make sense to handle the zero case here and handle the min/maxWidth checks in putNumber, since those are specific to the number type?

Reply:
Hmm, currently for unsigned ints, the minWidth check is the nBits == 0 check, so I think moving it out to the unparse would increase duplication since the UE uses the same outOfRangeFmtStr that the SDW uses, reducing duplication.

Reply:
Gotcha, as long as we make sure all the putNumber calls are doing the correct range checking and throwing a UE when it's out of range that's fine. Feels like float, double, and decimal aren't currently erroring on zero, but maybe zero is impossible for float/double, so only decimal needs it?